### PR TITLE
[AJ-267] scala 2.13

### PR DESCRIPTION
**Pull request by MatthewBemis**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala [hunk 1]
```diff
@@ -40,10 +40,14 @@ class WorkspaceServiceSpec extends BaseServiceSpec with BeforeAndAfterEach {
         status
       }
 
-      val tsvReturnString = List(
+      val tsvReturnStringOld = List(
```
Comment (MatthewBemis) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala:43]: todo remove

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala [hunk 2]
```diff
@@ -40,10 +40,14 @@ class WorkspaceServiceSpec extends BaseServiceSpec with BeforeAndAfterEach {
...
         List("workspace:f", "d", "b", "a", "e", "c").mkString("\t"),
         List("[\"v6\",999,true]", "escape quo\\\"te", 1.23, "true", "v1", "").mkString("\t")).mkString("\n")
 
+      val tsvReturnString = List(
+        List("workspace:e", "d", "b", "c", "a", "f").mkString("\t"),
+        List("v1", "escape quo\\\"te", 1.23, "", "true", "[\"v6\",999,true]").mkString("\t")).mkString("\n")
```
Comment (MatthewBemis) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala:45]: the column order returned by this code changed, not sure why- maybe some underlying implementation to collections?

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSpec.scala [hunk 3]
```diff
@@ -458,7 +458,7 @@ class LibraryServiceSpec extends BaseServiceSpec with AnyFreeSpecLike with Libra
         )))
         val expected = Document(testUUID.toString, Map(
           orspIdAttribute -> AttributeString("MOCK-111"),
-          consentCodesAttributeName -> AttributeValueList(Seq(AttributeString("NCU"), AttributeString("NCTRL"))),
+          consentCodesAttributeName -> AttributeValueList(Seq(AttributeString("NCTRL"), AttributeString("NCU"))),
```
Comment (MatthewBemis) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryServiceSpec.scala:461]: order changed, see https://github.com/broadinstitute/firecloud-orchestration/pull/930/files#r851575080

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupportSpec.scala [hunk 4]
```diff
@@ -38,7 +38,7 @@ class DataUseRestrictionSupportSpec extends AnyFreeSpec with Matchers with DataU
             irbRequired = true,
             prefix = Some("blah"))
 
-          val expected = Map("blahconsentCodes" -> Array("NPU","RS-G","NCU","HMB","RS-FM","NCTRL","RS-PD","IRB","NAGR","GRU","NMDS","DS:Ebola hemorrhagic fever","DS:hematologic cancer").toJson,
+          val expected = Map("blahconsentCodes" -> Array("NAGR","NMDS","NCTRL","RS-G","GRU","RS-PD","NCU","RS-FM","NPU","HMB","IRB","DS:Ebola hemorrhagic fever","DS:hematologic cancer").toJson,
```
Comment (MatthewBemis) [inline src/test/scala/org/broadinstitute/dsde/firecloud/service/DataUseRestrictionSupportSpec.scala:41]: order changed, see https://github.com/broadinstitute/firecloud-orchestration/pull/930/files#r851575080

File: src/main/scala/org/broadinstitute/dsde/firecloud/model/OrchMethodRepository.scala [hunk 5]
```diff
@@ -6,7 +6,7 @@ import org.broadinstitute.dsde.rawls.model.{AgoraMethod, DockstoreMethod, Method
 
 import scala.util.Try
 
-object MethodRepository {
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/firecloud/model/OrchMethodRepository.scala:9]: This seemed to conflict with a `MethodRepository` class included in rawls-model, which caused tons of issues with the implicits contained in this file.

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala (hunk 2 above)
Comment (MatthewBemis) [reply src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala:45]: @davidangb can you think of any problematic side effects of this?

**Review by Qi77Qi — APPROVED** [review]
🎉  lgtm other than the collection order change is a bit strange. Wonder if not using `Seq` will help (if order matters)

File: src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala (hunk 2 above)
Comment (davidangb) [reply src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala:45]: looking at this now - the ordering changes are concerning on the surface but might be totally harmless.

Comment (davidangb) [reply src/test/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceServiceSpec.scala:45]: Using [Scastie](https://scastie.scala-lang.org/davidangb/JcM2HlKTT7GIPolxSQBvog/6), I've made a simple test case:
```scala
// use the "Build Settings" to swap between Scala 2.12.15 and 2.13.8, then run:

val mapWithTupleKeys = Map(
  ("default", "e") -> "",
  ("default", "d") -> "",
  ("default", "b") -> "",
  ("default", "c") -> "",
  ("default", "a") -> "",
  ("default", "f") -> ""
)

println(util.Properties.versionString + ": " + mapWithTupleKeys.map(_._1._2).mkString(","))

// results:
// Scala 2.12.15: f,d,b,a,e,c
// Scala 2.13.8: d,a,e,b,c,f

```

**Review by davidangb — APPROVED** [review]
I think [jenkins/ittests.sh](jenkins/ittests.sh) needs an update of its `SBT_IMAGE`.

The key-ordering changes make me uncomfortable … but see my test case, it really looks like a change in the underlying collection behavior, and we should never be relying on ordering of Map keys. I can't come up with a reason this would cause problems but let's keep an eye out.

other than that, :+1: and thank you so much!

**Pull request merged**